src/logic/basic.lean

```diff
@@ -976,6 +976,9 @@ by simp [@eq_comm _ a']
 theorem exists_comm {p : α → β → Prop} : (∃ a b, p a b) ↔ ∃ b a, p a b :=
 ⟨λ ⟨a, b, h⟩, ⟨b, a, h⟩, λ ⟨b, a, h⟩, ⟨a, b, h⟩⟩
 
+theorem exists_and_index {p q : Prop} {f : p ∧ q → Prop} : (∃ h, f h) ↔ ∃ hp hq, f ⟨hp, hq⟩ :=
```

Comment:
Do we have this lemma for prod already? Is the name consistent? I'd expect for to be called `and.forall`.

Reply:
I had a scan, and couldn't find it. I like your name suggestion.

Reply:
Renamed.